protostar/starknet_gateway/signer.py

```diff
@@ -0,0 +1,33 @@
+from dataclasses import dataclass
```

Comment:
these classes are only used in `signer_factory.py`

could these be moved there?